### PR TITLE
feat: Get config from parameter store and secrets manager

### PR DESCRIPTION
**Pull request by mattbell87**
## Checklist
- [ ] I'm using the correct PHP Version (8.1 for current, 7.4 for legacy).
- [ ] I've added comments to any new methods I've created or where else relevant.
- [ ] I've replaced magic method usage on DbService classes with the getInstance() static method.
- [ ] I've written any documentation for new features or where else relevant in the docs [repo](https://github.com/2pisoftware/cmfive-docs).

## Changelog

refs:
issues:

## Other Information

Docs pull request:

### CODE REVIEW

File: system/classes/Config.php [hunk 1]
```diff
@@ -338,6 +355,78 @@ public static function setFromS3Object(string $bucket, string $key): void
...
+     * is expected to be valid JSON. If the JSON decode fails the function will fall back to using setFromS3Object or
+     * an exception will be thrown.
+     *
+     * @param string $parameterName
+     * @param string $bucket
+     * @param string $key
+     * @return void
+     * @throws Exception
+     */
+    public static function setFromParameterStore(string $parameterName, string $bucket, string $key): void
```
Comment (adam-buckley) [inline system/classes/Config.php:370]: The S3 bucket and key should be optional with a check to see if they're set

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: system/classes/Config.php (hunk 1 above)
Comment (mattbell87) [reply system/classes/Config.php]: Done

File: system/classes/Config.php [hunk 2]
```diff
@@ -362,12 +361,12 @@ public static function setFromS3Object(string $bucket, string $key): void
...
      * @param string $parameterName
-     * @param string $bucket
-     * @param string $key
+     * @param string|null $bucket
+     * @param string|null $key
      * @return void
      * @throws Exception
      */
-    public static function setFromParameterStore(string $parameterName, string $bucket, string $key): void 
+    public static function setFromParameterStore(string $parameterName, ?string $bucket = null, ?string $key = null): void
```
Comment (adam-buckley) [inline system/classes/Config.php:369]: I just noticed, and I don't know if it's actually important but the "self" reference is usually all lowercase

**Review by adam-buckley — CHANGES_REQUESTED** [review]

File: system/classes/Config.php (hunk 2 above)
Comment (mattbell87) [reply system/classes/Config.php:369]: You are right, all fixed

**Review by adam-buckley — APPROVED** [review]

**Pull request merged**